### PR TITLE
Add A32 support in CI

### PR DESCRIPTION
**Pull request by Cuda-Chen**
Add A32 support in CI.

For @jonathanhue, if possible, kindly leave some suggestions as I have little experience on A32 binary running on Armv8-A.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -34,7 +34,11 @@ ifeq ($(processor),$(filter $(processor),aarch64 arm64))
 else ifeq ($(processor),$(filter $(processor),i386 x86_64))
     ARCH_CFLAGS = -maes -mpclmul -mssse3 -msse4.2
 else ifeq ($(processor),$(filter $(processor),arm armv7 armv7l))
-    ARCH_CFLAGS = -mfpu=neon
+	ifdef A32
+		ARCH_CFLAGS = -mcpu=cortex-a32 -mfpu=neon-fp-armv8
```
Comment (jserv) [inline Makefile:38]: Enforce consistent indention.

File: README.md [hunk 2]
```diff
@@ -126,6 +126,10 @@ or
 ```shell
 $ make CROSS_COMPILE=arm-linux-gnueabihf- check # ARMv7-A
 ```
+or
+``` shell
+$ make CROSS_COMPILE=arm-linux-gnueabihf- A32=1 check # ARMv8-A running in 32-bit mode, i.e., A32 instruction set
```
Comment (jserv) [inline README.md:131]: In this case, A32 is always 1, right?

Comment (Cuda-Chen) [reply README.md:131]: Correct. A32 always sets to 1 if we want to run test program compiled under A32 instruction set.

File: Makefile (hunk 1 above)
Comment (Cuda-Chen) [reply Makefile]: Fixed.

File: README.md (hunk 2 above)
Comment (jserv) [reply README.md:131]: If toolchain is set to `arm-linux-gnueabihf-`, `A32` should not appear since the former implies the latter.

Comment (Cuda-Chen) [reply README.md:131]: The goal of using `A32` for indicating A32 instruction set build is that we can not only test our code on AArch32 but also check whether the NEON optimization on A32 will create any impacts or not.

Nevertheless, we may set the `-mcpu` flag in other Makefile input parameters such as `CFLAGS`, and I consider I can start to implement from this.

File: README.md [hunk 3]
```diff
@@ -126,6 +126,13 @@ or
 ```shell
 $ make CROSS_COMPILE=arm-linux-gnueabihf- check # ARMv7-A
 ```
+or
+``` shell
+$ make \
+  CROSS_COMPILE=arm-linux-gnueabihf- \
+  ARCH_CFLAGS="-mcpu=cortex-a32 -mfpu=neon-fp-armv8" \
+  check # ARMv8-A running in 32-bit mode, i.e., A32 instruction set
```
Comment (jserv) [inline README.md:134]: Move the comment out of code block.

File: .github/workflows/github_actions.yml [hunk 4]
```diff
@@ -55,7 +55,12 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        arch_with_features: [{arch: armv7, feature: none}, {arch: aarch64, feature: none}, {arch: aarch64, feature: crypto+crc}]
+        arch_with_features: [
+          {arch: armv7, feature: none, arch_cflags: none},
```
Comment (jserv) [inline .github/workflows/github_actions.yml:59]: At least one feature should be specified. That is `simd` (or NEON), right?

Comment (Cuda-Chen) [reply .github/workflows/github_actions.yml:59]: Correct. 

Based on my experiment, `uraimo/run-on-arch-action` does not permit use to expand variables in `env` via any `if-else` expression. As a consequence, I use `none` to let `make` to fill the default value of `ARCH_CFLAGS` compile flags, just like the `FEATURE` variable in Makefile does.

Comment (jserv) [reply .github/workflows/github_actions.yml:59]: Add necessary comments to describe the features.

Comment (Cuda-Chen) [reply .github/workflows/github_actions.yml:59]: Comments to describe the usage of `ARCH_CLFAGS` are added in README.

**Comment by jserv** [comment]
Thank @Cuda-Chen for contributing!

**Pull request merged**